src/app-host/app-host.js

```diff
@@ -83,6 +84,10 @@ socket.on('init-telemetry', function (data) {
     telemetry.init(socket);
 });
 
+socket.on('init-xhr-proxy', function (data) {
+    xhrProxy.init(); 
```

Comment:
xhrProxy is needed only when app-host received this message. So maybe, we can load the module, at this point instead of at the top of the module.